nexus/src/app/background/tasks/multicast/members.rs

```diff
@@ -0,0 +1,1495 @@
+                    | StateTransition::NoChange => {
+                        processed += 1;
+                        debug!(
+                            opctx.log,
+                            "processed member state change";
+                            "member" => ?member,
+                            "group" => ?group,
+                            "transition" => ?transition
+                        );
+                    }
+                    StateTransition::NeedsCleanup => {
+                        processed += 1;
+                        debug!(
+                            opctx.log,
+                            "member marked for cleanup";
+                            "member" => ?member,
+                            "group" => ?group
+                        );
+                    }
+                },
+                Err(e) => {
+                    warn!(
+                        opctx.log,
+                        "failed to process member state change";
+                        "member" => ?member,
+                        "group" => ?group,
+                        "error" => %e
+                    );
+                }
+            }
+        }
+
+        Ok(processed)
+    }
+
+    /// Main dispatch function for processing member state changes.
+    ///
+    /// Routes to appropriate node based on member type.
+    async fn process_member_state(
+        &self,
+        opctx: &OpContext,
+        group: &MulticastGroup,
+        member: &MulticastGroupMember,
+        dataplane_client: &MulticastDataplaneClient,
+    ) -> Result<StateTransition, anyhow::Error> {
+        // For now, all members are instance-based, but this is where we'd
+        // dispatch to different processors for different member types
+        let processor = InstanceMemberProcessor;
+
+        match member.state {
+            MulticastGroupMemberState::Joining => {
+                processor
+                    .process_joining(
+                        self,
+                        opctx,
+                        group,
+                        member,
+                        dataplane_client,
+                    )
+                    .await
+            }
+            MulticastGroupMemberState::Joined => {
+                processor
+                    .process_joined(
+                        self,
+                        opctx,
+                        group,
+                        member,
+                        dataplane_client,
+                    )
+                    .await
+            }
+            MulticastGroupMemberState::Left => {
+                processor
+                    .process_left(self, opctx, group, member, dataplane_client)
+                    .await
+            }
+        }
+    }
+
+    /// Instance-specific handler for members in "Joining" state.
+    /// Handles sled_id updates and validates instance state before proceeding.
+    async fn handle_instance_joining(
```

Comment:
There are several independent database operations in this reconciler function. Something to think about here is what happens when multiple nexus instances hare in here concurrently, potentially with slightly different views of the world since instance state will evolve asynchronously from this reconciler.